src/shapes/ActiveSelection.ts

```diff
@@ -23,6 +33,36 @@ export class ActiveSelection extends Group {
     return true;
   }
 
+  /**
+   * @private
+   * @override we don't want the selection monitor to be active
+   */
+  __objectSelectionMonitor() {
+    //  noop
+  }
+
+  /**
+   * Adds objects with respect to {@link multiSelectionStacking}
+   * @param targets object to add to selection
+   */
+  multiSelectAdd(...targets: FabricObject[]) {
+    if (this.multiSelectionStacking === 'selection-order') {
+      this.add(...targets);
+    } else {
+      //  respect object stacking as it is on canvas
+      //  perf enhancement for large ActiveSelection: consider a binary search of `isInFrontOf`
+      targets.forEach((target) => {
+        const index = this._objects.findIndex((obj) => obj.isInFrontOf(target));
+        const insertAt =
+          index === -1
+            ? //  `target` is in front of all other objects
+              this.size()
+            : index;
+        this.insertAt(insertAt, target);
+      });
+    }
+  }
```

Comment:
why calling this `add` and not using the fact that this is a subclass of group, exactly to handle multi selections, and so its own add method is either super.add or this.insertAt?
I would gladly rename this in add. Maybe i ll merge this as it is, and i ll just open a PR with the things i wouldn't want to do

Reply:
thought of that as well.
wasn't sure what I/you think about that.
The reason I am not for is in case the dev wants to change the stack order, using `insertAt` is valuable so if we change it it will be annoying.

Reply:
but anyone can do `this._objects.insertAt` so I am ok with it

Reply:
do it

Reply:
so this will be add and insertAt will be left as is?

Reply:
i was just saying to rename multiSelectAdd in add.
Because this class has to handle only multi selection logic and nothing else. so seems legit that its own add is special.

Are you thinking that people may want to call add of an active selection directly to do things in the active selection outside the canvas flow?

Reply:
Yes
Perhaps
What do you think?